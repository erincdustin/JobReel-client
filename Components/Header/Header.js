import React from 'react';
import {Link} from 'react-router-dom';

import './Header.css';

class Header exteds React.Component {

  static contextType = UserContext

  handleLogOutClick = () => {
    console.log('this function calls for context logout')
  }

  renderLogoutLink() {
    return (
      <div className='nav'>
        <span>
          !!!Placeholder for user name!!!
        </span>
        <nav role ='navigation'>
          <Link 
            onClick={this.handleLogOutClick}
            to='/login'>
            Log Out
          </Link>
        </nav>
      </div>
    )
  }

  renderLoginLink() {
    return (
      <div className='nav'>
        <nav role='navigation'>
          <Link to ='/login'>Login</Link>
          {' '}
          <Link to='/register'>Sign Up</Link>
        </nav>
      </div>
    )
  }

  render() {
    return (
      <header className='header'>
        <h1 className='logo'>
          <Link to ='/'>
            JobReel
          </Link>
        </h1>
        <div className='conditional render based on auth'>
          {this.renderLoginLink}
          {this.renderLogoutLink}
        </div>

      </header>
    )
  }
}