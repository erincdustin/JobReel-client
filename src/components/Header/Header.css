.header {
  max-width: 100vw;
  background: #3a3846;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-gap: 16px;
  height: 75px;
  margin-left: 0%;
  position: relative;
}
.user-cog{
  color: white;
}

.fish{
  display: inline-block;
}

.app-name{
  display: inline-block;
}

.logo {
  grid-column-start: 1;
  grid-column-end: 4;
  text-decoration: none;
  margin: auto auto auto 20px;
  display: inline-block;
  font-family: 'Montserrat', sans-serif;
  font-size: 26px;
}

.logo > a > img {
  height: 50px;
  width: 50px;
}

.nav-links {
  grid-column-start: 10;
  grid-column-end: 13;
  font-family: 'Montserrat', sans-serif;
  margin: auto 20px auto auto;
  width: 100%;
  height: 100%;
  text-decoration: none;
  text-align: center;
}


.username {
  grid-column-start: 6;
  grid-column-end: 8;
  font-family: 'Montserrat', sans-serif;
  color: white;
  margin: auto;
  text-align: center;
  padding: 5px;
}

.nav-links a {
  text-decoration: none;
  display: block;
  text-align: right;
  padding-right: 15px;
  padding-top: 7.5px;
  font-size: 16px;
  margin: 5px;
}


@media screen and (max-width: 870px) {
  .header {
    grid-template-columns: 1fr 1fr 1fr;
    height: 75px;
  }

  .username {
    grid-column-start: 2;
    grid-column-end: 3;
    margin: auto;
    text-align: center;
    color: white;
  }

  .nav-links {
    grid-column-start: 3;
    grid-column-end: 4;
    margin: auto;
    text-align: center;
  }

  .logo {
    grid-column-start: 0;
    grid-column-end: 2;
    margin: auto auto auto 20px;
    font-family: 'Montserrat', sans-serif;
    font-size: 20px;
  }

  .nav-links a{
    display: block;
    text-align: right;
    padding-right: 15px;
    padding-top: 10px;
    font-size: 13px;
    margin: 5px;
  }
}

@media screen and (max-width: 400px) {
  .logo {
    font-size: 16px;
    margin-left: 10px;
  }

  .nav-links  a{
    display: block;
    text-align: right;
    padding-right: 10px;
    font-size: 12px;
    margin: 5px;
  }
}