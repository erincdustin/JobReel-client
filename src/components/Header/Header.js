import React from 'react';
import {Link} from 'react-router-dom';
import TokenService from '../../services/token-service';
import './Header.css';
import JobReelContext from '../../context/JobReelContext';
import './Header.css';
import logo from '../../assests/jobreellogo200.png'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

class Header extends React.Component {

  static contextType = JobReelContext

  handleLogOutClick = () => {
    this.context.processLogout()
  }

  renderLogoutLink() {
    return (
      <div className='nav'>
        <nav role ='navigation'>
          <Link 
            onClick={this.handleLogOutClick}
            to='/login'>
            Log Out
          </Link>
        </nav>
      </div>
    )
  }

  renderLoginLink() {
    return (
      <div className='nav'>
        <nav role='navigation'>
          <Link to ='/login'>Login</Link>
          {' '}
          <Link to='/'>Sign Up</Link>
        </nav>
      </div>
    )
  }

  renderUsername() {
    if (this.context.user.username) {
      return (
        <>
          Signed in as  {this.context.user.username.toUpperCase()} <Link to='/profile'><FontAwesomeIcon icon='user-cog'/></Link>
        </>
      )
    } else {
      return (
        <>
        </>
      )
    }
  }

  render() {
    return (
      <header className='header'>
        <div className='logo'>
          <Link to='/dashboard'>
            <img src={logo} alt='jobreel-logo'/>
          </Link>
        </div>
        <div className='username'>
            {this.renderUsername()}
        </div>
          {TokenService.hasAuthToken()
          ? this.renderLogoutLink()
          : this.renderLoginLink()}

      </header>
    )
  }
}

export default Header;