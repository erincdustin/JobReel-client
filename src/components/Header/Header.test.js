import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import PropTypes from 'prop-types'
import toJson from 'enzyme-to-json' 
import {BrowserRouter} from 'react-router-dom'
import Header from './Header'

class ContextProvider extends React.Component {
  static childContextTypes = {
    myContext: PropTypes.object,
  }

  getChildContext = () => ({
    myContext: {
      user: {
        username: 'testName'
      }
    }
  })

  render () {
    return this.props.children
  }
}

describe('Header smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><Header /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('Header snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<ContextProvider><Header /></ContextProvider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})