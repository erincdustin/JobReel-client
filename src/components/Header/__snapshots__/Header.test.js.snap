// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot test renders the UI as expected 1`] = `
<header
  className="header"
>
  <div
    className="logo"
  >
    <Link
      to="/dashboard"
    >
      <img
        alt="jobreel-logo"
        src="jobreellogo200.png"
      />
    </Link>
  </div>
  <div
    className="nav"
  >
    <nav
      role="navigation"
    >
      <Link
        to="/login"
      >
        Login
      </Link>
       
      <Link
        to="/"
      >
        Sign Up
      </Link>
    </nav>
  </div>
</header>
`;
